File: post/models/post.js
const mongoose = require("mongoose");

const postSchema=new mongoose.Schema({user: {
    type: mongoose.Schema.Types.ObjectId,
    ref: "user",
  },
  text: {
    type: String,
    required: true,
  },
  image:{
    type: String,
  },
  name: { type: String },
  avatar: { type: String },
  likes: [
    {
      user: {
        type: mongoose.Schema.Types.ObjectId,
        ref: "user",
      },
    },
  ],
  comments: [
    {
      user: {
        type: mongoose.Schema.Types.ObjectId,
        ref: "user",
      },
      text: { type: String, required: true },
      avatar: { type: String },
      date: { type: Date, default: Date.now },
    },
  ],
  date: { type: Date, default: Date.now },
})

module.exports=mongoose.model("post",postSchema)

File: mobile/lib/models/post.dart
import 'dart:html';

import 'package:mobile/models/like.dart';
import 'package:mobile/models/comment.dart';

class Post {
  final String id;
  final String user;
  final String text;
  final String image;
  final List<Like> likes;
  final List<Comment> comments;
  final DateTime date;
  final int v;

  Post(
      {required this.id,
      required this.user,
      required this.text,
      required this.image,
      required this.likes,
      required this.comments,
      required this.date,
      required this.v});
}
